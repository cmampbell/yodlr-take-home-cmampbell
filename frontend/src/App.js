import React from 'react';
import { Outlet } from 'react-router-dom'
import NavBar from './NavBar'
import './App.css';

function App() {
  return (
    <div className="App">
      <NavBar/>
      <Outlet/>
    </div>
  );
}

export default App;
