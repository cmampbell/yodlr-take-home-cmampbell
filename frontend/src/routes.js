import App from "./App";
import SignupForm from "./SignupForm";
import AdminPage from "./AdminPage";
import HomePage from "./HomePage"

const baseURL = 'http://localhost:3001'

const routes = [
    {
        element: <App/>,
        children: [
            {
                path: '/',
                element: <HomePage />
            },
            {
                path: '/signup',
                element: <SignupForm/>,
            },
            {
                path: '/admin',
                element: <AdminPage />,
                loader: async () => {
                    return await fetch(`${baseURL}/users`);
                }
            }
        ]
    }
];

export default routes;