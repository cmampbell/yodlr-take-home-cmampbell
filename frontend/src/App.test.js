import renderWithRouter from "./renderWithRouter";
import App from './App';
import routes from './routes'
import { fireEvent, waitFor } from "@testing-library/react";
import { screen } from "@testing-library/react";

describe('Tests for App', () => {
  it('should render without crashing', () => {
    renderWithRouter(<App />);
  });

  it('should match snapshot', () => {
    const { asFragment } = renderWithRouter(<App />);

    expect(asFragment()).toMatchSnapshot();
  })
})

describe('Tests for navigation', () => {
  it('should have links to move between pages', async () => {
    const { queryByLabelText } = renderWithRouter(<App />, routes);

    fireEvent.click(screen.getByText('Sign Up'));

    waitFor(() => {
      expect(queryByLabelText('First Name')).toBeInTheDocument();
      expect(screen.getByText('Welcome to Yodlr!')).not.toBeInTheDocument();
    })
  })
})