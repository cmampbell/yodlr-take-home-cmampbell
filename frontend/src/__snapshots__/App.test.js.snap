// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="NavBar"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
        id="NavBar-Yodlr"
      >
        Yodlr
      </a>
      <a
        class=""
        href="/signup"
      >
        Sign Up
      </a>
      <a
        class=""
        href="/admin"
      >
        Admin Page
      </a>
    </div>
  </div>
</DocumentFragment>
`;
